baker/Deployer.ts

```diff
@@ -0,0 +1,374 @@
+
+        const scripts: any = {
+            file: makeScriptToDoFileStuff(ROOT, target, SYNC_TARGET, TMP_NEW),
+            yarn: makeScriptToDoYarnStuff(TMP_NEW),
+            adminServerStuff: makeScriptToDoQueueStuffDoFileStuffDoAdminServerStuff(
+                target,
+                ROOT_TMP,
+                TMP_NEW,
+                FINAL_TARGET
+            ),
+            bake: `pm2 stop ${target}-deploy-queue
+# Static build to update the public frontend code
+cd ${FINAL_TARGET}
+node itsJustJavascript/baker/bakeSite.js`,
+            deploy: makeScriptToDeployToNetlifyDoQueue(
+                target,
+                gitEmail,
+                gitName,
+                FINAL_TARGET
+            ),
+        }
+
+        Object.keys(scripts).forEach((name) => {
+            const fullName = `${name}${TEMP_DEPLOY_SCRIPT_SUFFIX}`
+            const localPath = `${owidGrapherRootDir}/${fullName}`
+            fs.writeFileSync(localPath, scripts[name], "utf8")
+            fs.chmodSync(localPath, "755")
+        })
+
+        await this.copyLocalRepoToServerTmpDirectory()
+
+        for await (const name of Object.keys(scripts)) {
+            await this.runAndStreamScriptOnRemoteServerViaSSH(
+                `${SYNC_TARGET}/${name}${TEMP_DEPLOY_SCRIPT_SUFFIX}`
+            )
+        }
+    }
+
+    printAndExit(message: string) {
+        // eslint-disable-next-line no-console
+        console.log(message)
+        process.exit()
+    }
+
+    private async ensureTmpDirExistsOnServer() {
+        const { sshHost } = this
+        const { owidUserHomeTmpDir } = this.pathsOnTarget
+        await execWrapper(`ssh ${sshHost} mkdir -p ${owidUserHomeTmpDir}`)
+        this.progressBar.tick({
+            name: `✅ 📡 finished ensuring ${owidUserHomeTmpDir} exists on ${sshHost}`,
+        })
+    }
+
+    private async copyLocalRepoToServerTmpDirectory() {
+        const { owidGrapherRootDir } = this.options
+        const { rsyncTargetDir } = this.pathsOnTarget
+        const RSYNC = `rsync -havz --no-perms --progress --delete --delete-excluded --exclude-from=${owidGrapherRootDir}/.rsync-ignore`
+        await execWrapper(
+            `${RSYNC} ${owidGrapherRootDir}/ ${this.sshHost}:${rsyncTargetDir}`
+        )
+        this.progressBar.tick({
+            name: `✅ 📡 finished rsync of ${owidGrapherRootDir} to ${this.sshHost} ${rsyncTargetDir}`,
+        })
+    }
+
+    private async runAndStreamScriptOnRemoteServerViaSSH(path: string) {
+        // eslint-disable-next-line no-console
+        console.log(`📡 Running ${path} on ${this.sshHost}`)
+        const params = [`-t`, this.sshHost, "bash -e", path]
+        const child = spawn(`ssh`, params)
+
+        child.stdout.on("data", (data) => {
+            // eslint-disable-next-line no-console
+            console.log(data.toString())
+        })
+
+        child.stderr.on("data", (data) => {
+            // eslint-disable-next-line no-console
+            console.log(data.toString())
```

Comment:
probably better to use `console.error` here to differentiate that output?

Reply:
Hmm, probably? I know some stuff is printing to stderr that maybe shouldn't. (I think progress bar? forget which). I am not crazy about `stderr` and `stdout` (just like `null` and `undefined`, really what you want is `N` standard channels).